src/plugins/data/server/search/strategies/ese_search/ese_search_strategy.ts

```diff
@@ -45,56 +45,93 @@ export const enhancedEsSearchStrategyProvider = (
   usage?: SearchUsage,
   useInternalUser: boolean = false
 ): ISearchStrategy<IEsSearchRequest<IAsyncSearchRequestParams>> => {
-  function cancelAsyncSearch(id: string, esClient: IScopedClusterClient) {
+  function cancelAsyncSearch(id: string, { esClient }: SearchStrategyDependencies) {
     const client = useInternalUser ? esClient.asInternalUser : esClient.asCurrentUser;
     return client.asyncSearch.delete({ id });
   }
 
-  function asyncSearch(
+  async function asyncSearchStatus(
     { id, ...request }: IEsSearchRequest<IAsyncSearchRequestParams>,
     options: IAsyncSearchOptions,
+    { esClient }: Pick<SearchStrategyDependencies, 'esClient'>
+  ) {
+    const client = useInternalUser ? esClient.asInternalUser : esClient.asCurrentUser;
```

Comment:
Reading the documentation about the async search status they stated that:
>If the Elasticsearch security features are enabled, the access to the get async search status API is restricted to the [monitoring_user role](https://www.elastic.co/guide/en/elasticsearch/reference/8.12/built-in-roles.html).

But instead the other async search API is restricted only to the user or API key that submitted it.
> If the Elasticsearch security features are enabled, the access to the results of a specific async search is restricted to [the user or API key that submitted it](https://www.elastic.co/guide/en/elasticsearch/reference/8.12/security-limitations.html#can-access-resources-check).

Do we already check this user role before using the async search status?

Reply:
This was actually picked up by the functional tests (we had a bunch of failures). We worked with the ES team to update the required privileges to allow the user who created the async search to get the status of the results: https://github.com/elastic/elasticsearch/pull/106638